src/docfx/lib/process/ProcessUtility.cs

```diff
@@ -73,19 +80,19 @@ static void PipeStream(TextReader input, TextWriter output1, TextWriter? output2
 
     /// <summary>
     /// Reads the content of a file.
-    /// When used together with <see cref="WriteFile(string,string)"/>, provides inter-process synchronized access to the file.
+    /// When used together with <see cref="WriteJsonFile(string,string)"/>, provides inter-process synchronized access to the file.
     /// </summary>
     public static T ReadJsonFile<T>(string path) where T : class, new()
     {
-        var content = "";
+        byte[] bytes;
         using (InterProcessMutex.Create(path))
         {
-            content = File.ReadAllText(path);
+            bytes = File.ReadAllBytes(path);
         }
 
         try
         {
-            return JsonUtility.DeserializeData<T>(content, new FilePath(path));
+            return JsonSerializer.Deserialize<T>(bytes, s_jsonOptions) ?? new();
```

Comment:
Is `byte[]` better than `string` here?

Reply:
Absolutely! `byte[]` takes less memory and `JsonSerializer` is internally back by `Utf8JsonReader` which works directly on bytes not strings.